storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -119,24 +131,45 @@ func (s *fileSystemSource) enqueueReaders(
 	readersCh chan<- shardReaders,
 ) {
 	for shard, tr := range shardsTimeRanges {
-		files := fs.ReadInfoFiles(s.fsopts.FilePathPrefix(),
+		readInfoFilesResults := fs.ReadInfoFiles(s.fsopts.FilePathPrefix(),
 			namespace, shard, s.fsopts.InfoReaderBufferSize(), s.fsopts.DecodingOptions())
-		if len(files) == 0 {
+
+		if len(readInfoFilesResults) == 0 {
 			// Use default readers value to indicate no readers for this shard
 			readersCh <- newShardReaders(shard, tr, nil)
 			continue
 		}
 
-		readers := make([]fs.FileSetReader, 0, len(files))
-		for i := 0; i < len(files); i++ {
+		readers := make([]fs.FileSetReader, 0, len(readInfoFilesResults))
+		for i := 0; i < len(readInfoFilesResults); i++ {
+			result := readInfoFilesResults[i]
+			if result.Err.Error() != nil {
+				s.log.WithFields(
+					xlog.NewField("shard", shard),
+					xlog.NewField("namespace", namespace.String()),
+					xlog.NewField("error", result.Err.Error()),
+					xlog.NewField("timeRange", tr),
+					xlog.NewField("filepath", result.Err.Filepath()),
+				).Error("unable to read info files in enqueueReaders")
+				continue
+			}
+			info := result.Info
+
 			r, err := readerPool.get()
 			if err != nil {
 				s.log.Errorf("unable to get reader from pool")
 				readersCh <- newShardReadersErr(shard, tr, err)
 				continue
 			}
-			t := xtime.FromNanoseconds(files[i].Start).Round(0).UTC()
-			if err := r.Open(namespace, shard, t); err != nil {
+			t := xtime.FromNanoseconds(info.BlockStart).Round(0).UTC()
```

Comment:
nit: Hm I know you didn't write this code, but I think `FromNanoseconds` calls `time.Unix(0, nanos)` which I think is guarenteed to treat the nanos as UTC and since we don't put times in hash maps anymore its safe to just keep time structs around in the local timezone.  So perhaps remove the `Round` and `UTC` calls here?

Reply:
done